vital/algo/resection_camera.h

```diff
@@ -0,0 +1,74 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief resection_camera algorithm definition
+ */
+
+#ifndef VITAL_ALGO_RESECTION_CAMERA_H_
+#define VITAL_ALGO_RESECTION_CAMERA_H_
+
+#include <vital/vital_config.h>
+
+#include <vector>
+#include <memory>
+
+#include <vital/algo/algorithm.h>
+#include <vital/types/feature_set.h>
+#include <vital/types/match_set.h>
+#include <vital/types/camera_perspective.h>
+
+namespace kwiver {
+namespace vital {
+namespace algo {
+
+/// An abstract base class to resection a camera using 3D feature
+/// and point projection pairs.
+
+class VITAL_ALGO_EXPORT resection_camera
+  : public kwiver::vital::algorithm_def<resection_camera>
+{
+public:
+  /// Return the name of this algorithm
+  static std::string static_type_name() { return "resection_camera"; }
+
+  /// Estimate camera parameters from 3D points and their corresponding projections
```

Comment:
Missing period, and too long:
```suggestion
  /// Estimate camera parameters from 3D points and their corresponding
  /// projections.
```